kong/plugins/aws-lambda/handler.lua

```diff
@@ -19,20 +31,62 @@ function AWSLambdaHandler:new()
   AWSLambdaHandler.super.new(self, "aws-lambda")
 end
 
-local function retrieve_parameters()
-  ngx_req_read_body()
-
-  return utils.table_merge(ngx_req_get_uri_args(), public_utils.get_body_args())
-end
-
 function AWSLambdaHandler:access(conf)
   AWSLambdaHandler.super.access(self)
 
-  local bodyJson = cjson.encode(retrieve_parameters())
+  local upstream_body = new_tab(0, 6)
+
+  do
+    ngx_req_read_body()
+
+    local body_args, err_code, body_raw = public_utils.get_body_info()
+    if err_code == public_utils.req_body_errors.unknown_ct then
+      -- don't know what this body MIME type is, base64 it just in case
+      body_raw = ngx_encode_base64(body_raw)
```

Comment:
> I'm wondering if it's worse to insert property indicates that body has been base64 encoded (when it's done in plugin) into request to lambda

Yes, I was having the same thought. Something like `upstream_body.request_body.base64 = true` maybe? We could also *not* send what we think is binary data to the function - until someone comes to us with a good-case for that? Do you already have such a use-case for processing binary data in Lambda functions?